### DISCUSSION TITLE
Non-symmetric normalisation for directed graphs GNN

### DISCUSSION THREAD

**Discussion by seven-down** [pyg-team/pytorch_geometric#8228] [Q&A]
I have a set of directed graphs and have read that non-symmetric normalisation (D^-1A) of the adjacency matrix is preferable when building a GNN. There doesn't appear to be any transformers available for this normalisation and the GCNConv layer normalizes symmetrically (D^-1/2AD^-1/2).

Any advice or resources would be greatly appreciated :)

**Comment by EdisonLeeeee** [comment]
I think part of the code in [get_laplacian](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/get_laplacian.py) is what you are looking for.

**Reply by rusty1s** [reply]
In addition, `D^-1 A` can be simply achieved via `aggr="mean"` in the message passing layers, so there is no good reason to pre-compute this in the first place.

**Reply by seven-down** [reply]
Many thanks for the reply! Just to clarify, using aggr="mean" in the message passing layers does this need to be applied to each layer or only the first convolutional layer?

**Reply by EdisonLeeeee** [reply]
In your case it should be applied to each layer, I think.

**Reply by seven-down** [reply]
That makes sense, thank you very much for the help :)
